src/blk/spdk/NVMEDevice.cc

```diff
@@ -479,9 +505,12 @@ static bool probe_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid, st
 {
   NVMEManager::ProbeContext *ctx = static_cast<NVMEManager::ProbeContext*>(cb_ctx);
 
+  opts->io_queue_size = UINT16_MAX;
+  opts->io_queue_requests = UINT16_MAX;
+  opts->keep_alive_timeout_ms = nvme_ctrlr_keep_alive_timeout_in_ms;
+
   if (trid->trtype != SPDK_NVME_TRANSPORT_PCIE) {
-    dout(0) << __func__ << " only probe local nvme device" << dendl;
-    return false;
+    return true;
```

Comment:
why? IIUC, this change practically disables the spdk backend from connecting to NVMe devices.

if we just want to match the transport with its id, probably we can instead drop the check as a whole.

Reply:
For NON-PCIE transport, we do not need do any PCI match. Because, we will always pass the trid related info in spdk_nvme_probe. So it means that the checking for non-pcie transport is not needed.

Reply:
One more thing. This change just to handle non PCIe device part. For PCIe device, it will still compare the trid.  So it will not affect the PCIe path.

Reply:
i see. just wanted to help myself understand the check better. after this change, we practically have:
```py
local_pcie_addr = None
try:
  local_pcie_addr = spdk.pci_addr_parse(trid.traddr)
except:
  pass
# we only probe for remote transport
if local_pcie_addr:
  trid_to_probe = None
else:
  trid_to_probe = trid
spdk_nvme_probe(trid_to_probe, ctxt, probe_cb, attach_cb)
# ...
def probe_cb(ctx, transport_id):
  if transport_id.type == TRANSPORT_PCIE:
    # need to take a closer look if it is a PCIe transport
    return ctx.transport_id == transport_id
  else:
    # okay, trid is passed to spdk_nvme_probe() for non-pcie,
    # so the found transport must be the one we specified.
    assert(ctx.transport_id == transport_id)
    return true
```

so i'd propose two changes:

1. add comment to explain the assumption. IMHO, "only probe local nvme device" would make more sense with more context. how about trading it for an `assert()` to document the connection between the non-null trid passed to `spdk_nvme_probe()`? something like
  ```c++
  if (trid->trtype == SPDK_NVME_TRANSPORT_PCIE) {
    if (spdk_nvme_transport_id_compare(&ctx->trid, trid)) {
      //
      return false;
    } else {
      return true;
    }
  } else {
    // for non-pcie devices, should always match the specified trid
    assert(!spdk_nvme_transport_id_compare(&ctx->trid, trid));
    return true;
  }
  ```
2. restructure `probe_cb()` to improve the readability.

Reply:
ack